src/cargo/sources/config.rs

```diff
@@ -136,7 +170,8 @@ impl<'gctx> SourceConfigMap<'gctx> {
         debug!("loading: {}", id);
 
         let Some(mut name) = self.id2name.get(&id) else {
-            return id.load(self.gctx, yanked_whitelist);
+            let src = id.load(self.gctx, yanked_whitelist)?;
+            return self.overlay_source(id, src, yanked_whitelist);
```

Comment:
We only call `overlay_source` on the result of a `load`

Should we name it `fn load` and perform the load and apply the overlay?  I worry that the current structure could make it easy to overlook applying the overlay while this pattern would make it the default to go through `Self::load` and we'd be less likely to introduce bugs to this

Reply:
Unfortunately we're already in a `fn load`. I renamed `overlay_source` to `load_and_overlay`, and modified it to do the load and the overlay. I think this is an improvement but not sure it it really addresses the concern.